cvat-ui/src/actions/annotation-actions.ts

```diff
@@ -1075,7 +1076,23 @@ export function updateAnnotationsAsync(statesToUpdate: any[]): ThunkAction {
             }
 
             const promises = statesToUpdate.map((objectState: any): Promise<any> => objectState.save());
-            const states = await Promise.all(promises);
+            let results = await Promise.allSettled(promises);
+            results = results.map((s, index) => {
+                if (s.status === 'fulfilled') {
+                    return Promise.resolve(s.value);
+                }
+                if (s.reason instanceof IncorrectUpdateError) {
+                    dispatch({
+                        type: AnnotationActionTypes.INCORRECT_ANNOTATIONS_UPDATE,
+                        payload: {
+                            error: s.reason,
+                        },
+                    });
+                    return statesToUpdate[index].delete(frame);
```

Comment:
Why we delete objects we tried to update? (as I understand it is a case when we totally erase a mask during editing) 
They should not be updated if it leads to their incorrect state

Try to create a bounding box and move it out of frame, it will be returned to the previous position automatically. I expect something like this for masks

Reply:
Fixed, now if mask becomes empty it is not updated